src/Microsoft.AspNetCore.OData/Extensions/HttpRequestExtensions.cs

```diff
@@ -264,8 +264,10 @@ public static ETag<TEntity> GetETag<TEntity>(this HttpRequest request, EntityTag
         /// </summary>
         /// <param name="request">The request on which to base the next page link.</param>
         /// <param name="pageSize">The number of results allowed per page.</param>
+        /// <param name="instance">Object which can be used to generate the skiptoken value.</param>
+        /// <param name="objectToSkipTokenValue">Function that takes in the last object and returns the skiptoken value string.</param>
         /// <returns>A next page link.</returns>
-        public static Uri GetNextPageLink(this HttpRequest request, int pageSize)
+        public static Uri GetNextPageLink(this HttpRequest request, int pageSize, object instance = null, Func<object,string> objectToSkipTokenValue = null)
         {
```

Comment:
I think that adding optional params to an extension method is a breaking change that requires recompilation of the app calling the library. We probably need to expose both overloads.